### PR TITLE
Freon fire update

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Freon fire now has a proper overlay and uses hotspots (i know)

## Why It's Good For The Game
Now you can see the fire approaching, also it looks cool

## Changelog

:cl:
expansion: freon fire uses hotspots to indicate fire tiles
/:cl:

### CODE REVIEW

**Comment by FernandoJ8** [comment]
It's so prettyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy

**Comment by LemonInTheDark** [comment]
Nice effect. Pending review.

**Comment by LemonInTheDark** [comment]
Code seems ok, but I don't really like the structure of the thing. I seem to recall the last time this was done you tried to do it as a subtype, did anything go wrong with that? is it because the hotspot can survive past the heating up reaction of the plasma fire or something?

**Comment by Ghilker** [comment]
> Code seems ok, but I don't really like the structure of the thing. I seem to recall the last time this was done you tried to do it as a subtype, did anything go wrong with that? is it because the hotspot can survive past the heating up reaction of the plasma fire or something?

the subtype wasn't really a subtype but a totally different path, and a subtype would too require a lot of code duplications and stuff like that, so it was best to just setup variables for the freon fire untill something else used a different format (maybe some reactions could use an "hotspot" for special effects

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -355,7 +366,7 @@
 		cached_gases[/datum/gas/water_vapor][MOLES] += burned_fuel / HYDROGEN_BURN_H2_FACTOR
 
 		energy_released += (FIRE_HYDROGEN_ENERGY_RELEASED * burned_fuel)
-		cached_results["fire"] += burned_fuel * 10
+		cached_results["fire"] += burned_fuel * 100
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/reactions.dm:369]: Excuse me?

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 2]
```diff
@@ -29,24 +29,31 @@
...
 			if(plas > 0.5 || trit > 0.5 || h2 > 0.5)
 				if(active_hotspot.temperature < exposed_temperature)
 					active_hotspot.temperature = exposed_temperature
 				if(active_hotspot.volume < exposed_volume)
 					active_hotspot.volume = exposed_volume
+			if(freon > 0.5)
+				if(active_hotspot.temperature > exposed_temperature)
+					active_hotspot.temperature = exposed_temperature
+				if(active_hotspot.volume < exposed_volume)
+					active_hotspot.volume = exposed_volume
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:45]: Doesn't this make freon an aggressively good firefighting tool?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
I'd be more happy with just a direct subtype, if only because what's here is a bit hard to parse the intent of, and I don't like the idea of making hotspot functioning even more shrouded in code

Not totally sure what that would imply though

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 2 above)
Comment (Ghilker) [reply code/modules/atmospherics/environmental/LINDA_fire.dm:45]: only if it overrides the hotspot with soh, but since the reaction doesn't set it to TRUE it shouldn't happen that easily

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (Ghilker) [reply code/modules/atmospherics/gasmixtures/reactions.dm:369]: had to increase this number by ten or the cold fire would die almost instantly, this doesn't change anything in the reaction itself

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/atmospherics/environmental/LINDA_fire.dm:45]: I don't really want freon to override standard fires in any case, think it would end up looking a bit weird. Why not make it an else if?

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:369]: Aight

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 3]
```diff
@@ -29,24 +29,31 @@
...
+	. = air_gases[/datum/gas/freon]
+	var/freon = . ? .[MOLES] : 0
 	if(active_hotspot)
 		if(soh)
 			if(plas > 0.5 || trit > 0.5 || h2 > 0.5)
 				if(active_hotspot.temperature < exposed_temperature)
 					active_hotspot.temperature = exposed_temperature
 				if(active_hotspot.volume < exposed_volume)
 					active_hotspot.volume = exposed_volume
+			if(freon > 0.5)
```
Comment (Ghilker) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:41]:
```suggestion
			else if(freon > 0.5)
```

**Comment by LemonInTheDark** [comment]
gahhhhhhhhhhhhhhhh
